RFS/src/test/java/com/rfs/integration/WorkerCompleteAfterErrorTest.java

```diff
@@ -0,0 +1,118 @@
+package com.rfs.integration;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Test case to verify worker will not deadlock and complete all work given failures
+ * during critical migration workflows.
+ */
+@Slf4j
+public class WorkerCompleteAfterErrorTest {
+
+
+    @ParameterizedTest(name = "Injected failure on {0} step")
+    @EnumSource(InjectedFailure.class)
+    public void SingleWorkerFailure(final InjectedFailure failure){
+        // Setup
+        setupCms();
+        setupMockSource();
+        setupMockTarget();
+
+        // PSUEDO: Create a 5 workers
```

Comment:
I'm not sure what you mean by hardware, these workers would be inside the same JVM as the test - makes for a fast and influenceable test scenario.

5 is arbitrary but I wanted more than 1 other worker to pose a basic level of contention _which shouldn't impact the outcome_.  Note; I'd imagine there will be more focused tests cases around work contention, but figure it would be good to have a basic practical version of it here.

Reply:
By "different hardware" I meant my old mac which can handle a few containers vs. the fancy new macs which can handle many.  Let's try to reduce the number of containers running to the bare minimum to facilitate folks working on potatoes where possible.  Sounds like `2` is the right answer here, then?

Reply:
@chelma  Oh, these workers are running inside the same JVM, so they should be tiny footprint - even a 5 year old laptop should easily handle 100s of the RFS workings running concurrently.  Disk IO _could_ be a bottleneck, but that is why a mock source cluster would be used.

Sharing CPU and CMS are the limiters in this test which it should easily handle.  If not we would be discovering something about a dependency or system architecture that we need to investigate for scaling.